hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/TestXceiverServerDomainSocket.java

```diff
@@ -0,0 +1,713 @@
+    DomainSocketFactory factory = DomainSocketFactory.getInstance(conf);
+    XceiverServerDomainSocket server = new XceiverServerDomainSocket(MockDatanodeDetails.randomDatanodeDetails(),
+        conf, null, readExecutors, metrics, factory);
+    DomainSocket sock = null;
+    try {
+      sock = factory.createSocket(readTimeout, writeTimeout, localhost);
+      assertTrue(sock.isOpen());
+      // send request
+      final DataOutputStream  outputStream = new DataOutputStream(sock.getOutputStream());
+      outputStream.writeShort(OzoneClientConfig.DATA_TRANSFER_VERSION);
+      outputStream.writeShort(GetBlock.getNumber());
+      getBlockRequest().writeDelimitedTo(outputStream);
+      outputStream.flush();
+
+      final DataInputStream inputStream = new DataInputStream(sock.getInputStream());
+      inputStream.readShort();
+    } catch (IOException e) {
+      assertTrue(e instanceof SocketTimeoutException);
+      assertTrue(e.getMessage().contains("read(2) error: Resource temporarily unavailable"));
+    } finally {
+      factory.close();
+      if (sock != null) {
+        try {
+          sock.close();
+        } catch (IOException e) {
+        }
+      }
+      server.stop();
+    }
+  }
+
+  @Test
+  @Timeout(30)
+  public void testReadTimeout() throws InterruptedException {
+    conf.set(OzoneClientConfig.OZONE_DOMAIN_SOCKET_PATH, new File(dir, "ozone-socket").getAbsolutePath());
+    conf.set(OzoneConfigKeys.OZONE_CLIENT_READ_TIMEOUT, "5s");
+    DomainSocketFactory factory = DomainSocketFactory.getInstance(conf);
+    XceiverServerDomainSocket server = new XceiverServerDomainSocket(MockDatanodeDetails.randomDatanodeDetails(),
+        conf, null, readExecutors, metrics, factory);
+    DomainSocket sock = null;
+    try {
+      server.start();
+      sock = factory.createSocket(readTimeout, writeTimeout, localhost);
+      assertTrue(sock.isOpen());
+
+      // server will close the DomainSocket if there is no message from client in OZONE_CLIENT_READ_TIMEOUT
+      Thread.sleep(5 * 1000);
+      // send request
+      final DataOutputStream outputStream = new DataOutputStream(sock.getOutputStream());
+      outputStream.writeShort(OzoneClientConfig.DATA_TRANSFER_VERSION);
+      outputStream.writeShort(GetBlock.getNumber());
+      getBlockRequest().writeDelimitedTo(outputStream);
+      outputStream.flush();
+    } catch (IOException e) {
+      assertTrue(e instanceof SocketException);
+      assertTrue(e.getMessage().contains("write(2) error: Broken pipe"));
+    } finally {
+      factory.close();
+      if (sock != null) {
+        try {
+          sock.close();
+        } catch (IOException e) {
+        }
+      }
+      server.stop();
+    }
+  }
+
+  @Test
+  @Timeout(30)
+  public void testMaxXceiverCount() throws IOException, InterruptedException {
+    conf.set(OzoneClientConfig.OZONE_DOMAIN_SOCKET_PATH, new File(dir, "ozone-socket").getAbsolutePath());
```

Comment:
`OzoneConfigKeys.OZONE_CLIENT_READ_TIMEOUT` is not explicitly set in this test case.

Should set it here as well?

Reply:
It has default value, which will be used. So we don't need to set it explicitly unless we want to change the value.